src/main/java/com/rsmart/kuali/kfs/cr/batch/service/impl/StaleCheckExtractServiceImpl.java

```diff
@@ -0,0 +1,329 @@
+            validationErrors.add("Check Total Amount is blank.");
+        }
+
+        if (ObjectUtils.isNull(checkReconciliation) || ObjectUtils.isNull(checkReconciliation.getCheckNumber())) {
+            validationErrors.add("Check Reconciliation does not exist in KFS.");
+        } else {
+            validateCheckReconciliation(staleCheckRow, checkReconciliation, validationErrors);
+        }
+
+        if (validationErrors.size() > 0) {
+            String validationErrorMessagesForRow = getValidationErrorsForRow(validationErrors);
+            LOG.warn("validateStaleCheckBatchRow:" + validationErrorMessagesForRow);
+            return validationErrorMessagesForRow;
+        }
+        return StringUtils.EMPTY;
+    }
+
+    protected String getValidationErrorsForRow(List<String> validationErrors) {
+        String failureMessage = "Stale Check Row could not be processed for the following reasons:";
+        for (int i=0; i<validationErrors.size(); ++i) {
+            failureMessage += " [" + Integer.toString(i + 1) + "] " + validationErrors.get(i);
+        }
+        return failureMessage;
+    }
+
+    protected void validateCheckReconciliation(StaleCheckBatchRow staleCheckRow, CheckReconciliation checkReconciliation, List<String> validationErrors) {
+        if (!StringUtils.equalsIgnoreCase(staleCheckRow.getBankCode(), checkReconciliation.getBankCode())) {
+            validationErrors.add("The Bank Code in the file (" + staleCheckRow.getBankCode() + ") row does not match the bank on the Check (" +
+                    checkReconciliation.getBankCode() + ").");
+        }
+
+        try {
+            Double checkTotalAmountParsed = NumberFormat.getNumberInstance(java.util.Locale.US).parse(staleCheckRow.getCheckTotalAmount()).doubleValue();
+            KualiDecimal checkTotalAmount = new KualiDecimal(checkTotalAmountParsed);
+            if (!checkTotalAmount.equals(checkReconciliation.getAmount())) {
+                validationErrors.add("The Total Check Amount in the file row (" + checkTotalAmount.toString() + ") does not match the amount on the Check (" +
+                        checkReconciliation.getAmount().toString() + ").");
+            }
+        } catch (java.text.ParseException ex) {
+            validationErrors.add("Error converting total amount to decimal [" + staleCheckRow.getCheckTotalAmount() + "] " + ex.toString());
+        }
+
+        if (StringUtils.equalsIgnoreCase(checkReconciliation.getStatus(), CRConstants.STALE)) {
+            validationErrors.add("Check Status is already STALE.");
+        }
+    }
+
+    protected byte[] safelyLoadFileBytes(String fileName) {
+        InputStream fileContents;
+        byte[] fileByteContent;
+        
+        try {
+            fileContents = new FileInputStream(fileName);
+        } catch (FileNotFoundException e1) {
+            LOG.error("Batch file not found [" + fileName + "]. " + e1.getMessage());

```

Comment:
the function name should be in these log statements, I would also do the error log statement by adding the actual error